### DISCUSSION TITLE
GUI - Change

### DISCUSSION THREAD

**Discussion by zrzynda** [oetiker/SmokePing#296] [Ideas]
I would like to change the loss color section from fractions (ex. 1/20) to % like 5%. Where is the source code to figure that out?

**Comment by Daxtorim** [comment]
What version are you using? Those fractions have been changed to ranges based on percentages (1%,5%,10%,25%,50%) a while ago:
![Screenshot](https://user-images.githubusercontent.com/72616153/118560097-401ca400-b769-11eb-85f8-1813908a2104.png)

The relevant code is here:
`%lc = (upper_boundary => ["displayed_text", "hex_colour"])`
https://github.com/oetiker/SmokePing/blob/dea1c9fd409e87ce6ad5d96eabc20a062f9f715b/lib/Smokeping.pm#L1269-L1285

**Reply by zrzynda** [reply]
I am sorry I am not a strong coder, I am on the network side of things. The follwoing link is where we have it deployed:

http://smokeping.cdsglobalcloud.com/smokeping/smokeping.cgi?target=Frequently-used_Websites

**Reply by zrzynda** [reply]
But for our cutomers it would make it easier to just show %'s and not fractions or ranges

**Reply by Daxtorim** [reply]
It looks like you are using the default 20 pings for everything, so it would be easier to specify your own legends in the config file rather than modifying source code.
Docs on the configuration: https://oss.oetiker.ch/smokeping/doc/smokeping_config.en.html
```
+ detail
++ loss_colors
# Loss Color  Legend
   0   26ff00 "0%"
   1   00b8ff "5%"
   2   0059ff "10%"
   5   7e00ff "25%"
  10   ff00ff "50%"
  15   ff5500 "75%"
  20   ff0000 "100%" 
```
